tlslite/tlsconnection.py

```diff
@@ -773,6 +776,9 @@ def _clientGetServerHello(self, settings, clientHello):
                         AlertDescription.illegal_parameter,
                         "Server selected ALPN protocol we did not advertise"):
                     yield result
+        if settings.useHeartbeatExtension and \
+                serverHello.getExtension(ExtensionType.heartbeat):
```

Comment:
what if server didn't allow us to send requests? (`peer_not_allowed_to_send`)
what if the value server sent is invalid?